examples/simple.rs

```diff
@@ -0,0 +1,236 @@
+    //
+    // These are all intended to be constructed rarely (perhaps even once per app (or once per thread))
+    // and provide caches and scratch space to avoid allocations
+    let mut font_cx = FontContext::default();
+    let mut layout_cx = LayoutContext::new();
+    let mut scale_cx = ScaleContext::new();
+
+    // Create a RangedBuilder
+    let mut builder = layout_cx.ranged_builder(&mut font_cx, &text, display_scale);
+
+    // Set default text colour styles (set foreground text color)
+    let brush_style = StyleProperty::Brush(foreground_color);
+    builder.push_default(&brush_style);
+
+    // Set default font family
+    let font_stack = FontStack::Source("system-ui");
+    let font_stack_style = StyleProperty::FontStack(font_stack);
+    builder.push_default(&font_stack_style);
+    builder.push_default(&StyleProperty::LineHeight(1.3));
+    builder.push_default(&StyleProperty::FontSize(16.0));
+
+    // Set the first 4 characters to bold
+    let bold = FontWeight::new(600.0);
+    let bold_style = StyleProperty::FontWeight(bold);
+    builder.push(&bold_style, 0..4);
+
+    // Build the builder into a Layout
+    let mut layout: Layout<Color> = builder.build();
+
+    // Perform layout (including bidi resolution and shaping) with start alignment
+    layout.break_all_lines(max_advance, Alignment::Start);
+    let width = layout.width().ceil() as u32;
+    let height = layout.height().ceil() as u32;
+
+    let mut img = RgbaImage::new(width + (padding * 2), height + (padding * 2));
+    for pixel in img.pixels_mut() {
+        *pixel = Rgba([
+            background_color.r,
+            background_color.g,
+            background_color.b,
+            255,
+        ]);
+    }
+
+    // Iterate over laid out lines
+    for line in layout.lines() {
+        // Iterate over GlyphRun's within each line
+        for glyph_run in line.glyph_runs() {
+            render_glyph_run(&mut scale_cx, &glyph_run, &mut img, padding);
+        }
+    }
+
+    // Write image to PNG file
+    let output_file = File::create("output.png").unwrap();
+    let png_encoder = PngEncoder::new(output_file);
+    img.write_with_encoder(png_encoder).unwrap();
+}
+
+fn render_glyph_run(
+    context: &mut ScaleContext,
+    glyph_run: &GlyphRun<Color>,
+    img: &mut RgbaImage,
+    padding: u32,
+) {
+    // Resolve properties of the GlyphRun
+    let mut run_x = glyph_run.offset();
+    let run_y = glyph_run.baseline();
+    let style = glyph_run.style();
+    let color = style.brush;
+
+    // Get the "Run" from the "GlyphRun"
+    let run = glyph_run.run();
+
+    // Resolve properties of the Run
+    let font = run.font();
+    let font_size = run.font_size();
+
+    // Get byte offset of font within collection (0 if font file is not a collection)
+    // TODO: expose directly in read-fonts
+    let font_collection_ref = font.data.as_ref();
+    let raw_font_ref = ReadFontsRef::from_index(font_collection_ref, font.index).unwrap();
```

Comment:
Swash has this method: https://docs.rs/swash/0.1.15/swash/struct.FontRef.html#method.from_index